src/github/taskDashboardWebviewProvider.ts

```diff
@@ -0,0 +1,1180 @@
+				break;
+			case 'switch-to-main':
+				await this.switchToMainBranch();
+				break;
+			case 'create-pull-request':
+				await this.createPullRequest();
+				break;
+			default:
+				await super._onDidReceiveMessage(message);
+				break;
+		}
+	}
+
+	private createTemporarySession(query: string, type: 'local' | 'remote'): string {
+		const tempId = this._taskManager.createTemporarySession(query, type);
+		// Immediately update the dashboard to show the temporary session
+		this.updateDashboard();
+		return tempId;
+	}
+	private removeTemporarySession(tempId: string): void {
+		this._taskManager.removeTemporarySession(tempId);
+		// Update dashboard to remove the temporary session
+		setTimeout(() => {
+			this.updateDashboard();
+		}, 500); // Small delay to allow real session to be created
+	}
+
+	private async handleChatSubmission(query: string): Promise<void> {
+		if (!query) {
+			return;
+		}
+
+		query = query.trim();
+
+		try {
+			// Check if user explicitly mentions @copilot for remote background session
+			if (query.startsWith('@copilot ')) {
+				return this.handleRemoteTaskSubmission(query);
+			}
+			// Check if user explicitly mentions @local for local workflow
+			else if (query.startsWith('@local ')) {
+				return this.handleLocalTaskSubmission(query);
+			}
+			// Determine if this is a general question or coding task
+			else {
+				if (await this.isCodingTask(query)) {
+					// Show quick pick to choose between local and remote work
+					const workMode = await this.showWorkModeQuickPick();
+					if (workMode === 'remote') {
+						return this.handleRemoteTaskSubmission(query);
+					} else if (workMode === 'local') {
+						return this.handleLocalTaskSubmission(query);
+					} else {
+						// User cancelled the quick pick
+						return;
+					}
+				} else {
+					// General question - Submit to ask mode
+					await vscode.commands.executeCommand('workbench.action.chat.open', {
+						query,
+						mode: 'ask'
+					});
+				}
+			}
+		} catch (error) {
+			Logger.error(`Failed to handle chat submission: ${error} `, DashboardWebviewProvider.ID);
+		}
+	}
+
+	private async handleLocalTaskSubmission(query: string) {
+		const cleanQuery = query.replace(/@local\s*/, '').trim();
+
+		const tempId = this.createTemporarySession(cleanQuery || query, 'local');
+		try {
+			await this.handleLocalTaskWithIssueSupport(cleanQuery || query);
+		} finally {
+			this.removeTemporarySession(tempId);
+		}
+	}
+
+	private async handleRemoteTaskSubmission(query: string) {
+		const cleanQuery = query.replace(/@copilot\s*/, '').trim();
+
+		const tempId = this.createTemporarySession(cleanQuery, 'remote');
+		try {
+			await this.createRemoteBackgroundSession(cleanQuery);
+		} finally {
+			// Remove temporary session regardless of success/failure
+			this.removeTemporarySession(tempId);
+		}
+	}
+
+	private async checkoutPullRequestBranch(pullRequest: { number: number; title: string; url: string }): Promise<void> {
```

Comment:
Can instead call this:

https://github.com/microsoft/vscode-pull-request-github/blob/e5ce10696b0010b836f8a9a96a4ecd6a33295a48/src/view/reviewManager.ts#L1083